extensions/amp-access/0.1/access-expr.js

```diff
@@ -44,3 +45,51 @@ export function evaluateAccessExpr(expr, data) {
     parser.yy = null;
   }
 }
+
+/**
+ * AmpAccessEvaluator evaluates amp-access expressions.
+ * It uses a cache to speed up repeated evals for the same expression.
+ */
+export class AmpAccessEvaluator {
+  /** */
+  constructor() {
+    /** @type {Object<string, boolean>} */
+    this.cache = null;
+
+    /** @type {JsonObject} */
+    this.lastData = null;
```

Comment:
Ivars should be suffixed with `_` and/or use `@private`.

Reply:
Thanks! I forgot about that 😬